internal/providers/vault/provider.go

```diff
@@ -89,3 +115,41 @@ func (p *Provider) GetValue(id string) (value []byte, err error) {
 	}
 	return
 }
+
+// GetValue obtains a value by id. The id should contain the path in Vault to the secret.
+// Prepend the path with `kv-v2:` when reading a secret from the KV v2 backend.
+// Append the path with `#fieldName` to fetch the value at given field; defaults to VaultDefaultField.
+// Example paths:
+//   - `kv/database/password` returns the value of field `value` in the secret object at given path.
+//   - `kv/database#password` returns the value of field `password` in the secret object at path `kv/database`.
+//   - `kv-v2:secret/data/database` returns the value of field `value` in secret object at path
+//     `secret/data/database`, unwrapping the secret data as required for KV v2 backend.
+//   - `kv-v2:secret/data/database#password` returns the value of field `password` in secret object at path
+//     `secret/data/database`, unwrapping the secret data as required for KV v2 backend.
+// Secrets in Vault are stored as objects in the shape of map[string]interface{}. In case of the KV v2 backend it is
+// wrapped in another object map[string]interface{} with a key/field `data` referencing the secret.
+// Note that in the Vault API the path to a KV v2 secret includes the `data` segment, while this is not required in
+// Vault CLI. Due to this provider using the Vault API client, the given id to GetValue must follow the conventions
+// of the Vault API path. Please see documentation of Vault for details.
+func (p *Provider) GetValue(id string) (value []byte, err error) {
```

Comment:
Will fix

Reply:
> Aren't named return values a common practice ?

Yes, a very common one but the problem is that for maintenance, they make things more difficult than the alternative. The problem is that as someone is going down the flow of the method reading the code, they will inevitably either miss the fact that some variable is a side-effect return value and/or they will need to go back up to the function definition to see its type. This kind of implicit return effect as well as back-and-forth reading has a real penalty for maintenance which we would like to avoid.

Reply:
Clear, will fix.